.travis.yml

```diff
@@ -8,47 +8,34 @@ matrix:
     - os: linux
       jdk: oraclejdk8
       language: scala
+      env:
+        - SCALANATIVE_GC=boehm
       before_install:
-        - sudo apt-get -qq update
-        - sudo sh -c "echo 'deb http://apt.llvm.org/precise/ llvm-toolchain-precise-3.7 main' >> /etc/apt/sources.list"
-        - sudo sh -c "echo 'deb http://apt.llvm.org/precise/ llvm-toolchain-precise main' >> /etc/apt/sources.list"
-        - wget -O - http://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
-        - sudo add-apt-repository --yes ppa:ubuntu-toolchain-r/test
-        - sudo apt-get -qq update
-        - |
-          sudo apt-get install -y \
-            clang++-3.7 \
-            llvm-3.7 \
-            llvm-3.7-dev \
-            llvm-3.7-runtime \
-            llvm-3.7-tool \
-            libgc-dev \
-            libunwind7-dev
+        - bash bin/travis_linux_setup.sh
 
-        # Install re2
-        # (libre2-dev) since Xenial (16.04 LTS) http://packages.ubuntu.com/xenial/libre2-dev
-        - sudo apt-get install -y make
-        - export CXX=clang++-3.7
-        - git clone https://code.googlesource.com/re2
-        - pushd re2
-        - git checkout 2017-03-01
-        - make -j4 test
-        - sudo make install prefix=/usr
-        - make testinstall prefix=/usr
-        - popd
+    - os: linux
+      jdk: oraclejdk8
+      language: scala
+      env:
+        - SCALANATIVE_GC=none
+      before_install:
+        - bash bin/travis_linux_setup.sh
+
+    - os: osx
+      osx_image: xcode7.3
+      env:
+        - SCALANATIVE_GC=boehm
+        - CLANG_FORMAT_PATH="/usr/local/opt/llvm@4/bin/clang-format"
+      before_install:
+        - bash bin/travis_mac_setup.sh
 
     - os: osx
       osx_image: xcode7.3
+      env:
+        - SCALANATIVE_GC=none
+        - CLANG_FORMAT_PATH="/usr/local/opt/llvm@4/bin/clang-format"
```

Comment:
`travis lint` complains that you can only define one env var per matrix entry. One option would be to to use something else than an env variable to trigger the formatting check or start using build stages https://docs.travis-ci.com/user/build-stages/warm-cache/